### ISSUE TITLE
IE errors

### ISSUE THREAD

**Issue by edsilv** [jmespath/jmespath.js#5]
Hi,

I've noticed that recently jmespath.js is having issues in IE. In > IE8 I'm getting `Invalid calling object`:

https://drive.google.com/file/d/0B64TWB9qdiOMZ2RKc2xLa2ItUzg/view?usp=sharing

In IE8 I'm getting `doesn't support property or method 'indexOf'`:

https://drive.google.com/file/d/0B64TWB9qdiOMaUNXMno1Y3hydFE/view?usp=sharing

The first error is related to this line:

https://github.com/jmespath/jmespath.js/blob/master/jmespath.js#L6

I can't see any recent commits where this has been changed...

I decided to try to work around this using lodash + browserify:

https://github.com/edsilv/jmespath.js

I've changed the `isArray` method to return `_.isArray`, `toString.call` to `_.toString`, and `Object.keys` to `_.keys`.

This is now working for my use cases in > IE8. There's still the problem with indexOf in IE8 which could probably be solved with _.indexOf.

I've strayed pretty far off the reservation though in terms of your code style/setup. I've switched to using gulp too which you may not be keen on.

**Comment by edsilv** [comment]
I've added _.indexOf and _.lastIndexOf and it now seems fine in IE8.

I've also added gulp-mocha and all tests are passing.

Let me know if you want me to create a pull request.

**Comment by jamesls** [comment]
Hi @edsilv, thanks for looking into this.

A pull request that fixes this issue would be great.  The one thing I'd be curious about is whether or not we need to pull in the whole lodash library, given the small size of this library.  How much bigger is the minified jmespath.js with lodash included?

If I'm reading the source properly, it looks like `_.isArray` is just `Array.isArray` (https://github.com/lodash/lodash/blob/master/lodash.js#L7759).  And instead of `toString.call`, would `Objects.prototype.toString.call` work?  From digging around a bit, it looks like IE 8 interprets `toString` as `window.toString`.

Also, I don't have a strong preference for gulp vs. grunt so I don't mind the switch.

**Comment by edsilv** [comment]
Hi,

I've updated it to only use the lodash modules it requires which has improved file size.

artifacts/jmespath.js was 465kb!, now 86kb
artifacts/jmespath.min.js was 74kb, now 28kb

This passes your test suite and works for my use cases, but I created a simple test app to try other queries and when trying your Washington Cities example on jmespath.org I get:

    TypeError: sort() expected argument 1 to be type array-string,array-number but received type undefined instead.

Not sure what the problem is.

Regarding _.isArray. It looks like it's checking for the existence of a native isArray and when not found applying a polyfill:

https://github.com/lodash/lodash/blob/npm-packages/lodash.isarray/index.js#L100

This is all just my attempt at a quick and dirty fix really. Lodash may not be the optimal solution.
